### PR TITLE
Loading config from string

### PR DESCRIPTION
**Pull request by slemrmartin**
new param `--source_settings` or `ENV['SOURCE_SETTINGS']` receives yaml string with configuration. 

`--config` and `--amounts` settings from file are loaded as before ('default' is used when not specified)
It means that `source_settings` replaces only config keys needed.

### CODE REVIEW

**Comment by slemrmartin** [comment]
@agrare please review

File: bin/mock-collector [hunk 1]
```diff
@@ -19,13 +19,15 @@ require require_dev_path if File.exist?(require_dev_path)
...
 # --amounts     (ENV["AMOUNTS"])     - file located in /config/amounts (you can use [large|default|small| amazon/default | openshift/small] etc.)
+# --configmap   (ENV["ENDPOINT_CONFIG"])   - config in YAMLs, higher priority than 'config' and 'amounts'
 def parse_args
   require 'optimist'
   opts = Optimist.options do
     opt :source, "Inventory Source UID", :type => :string, :required => ENV["SOURCE_UID"].nil?, :default => ENV["SOURCE_UID"]
     opt :ingress_api, "Hostname of the ingress-api route", :type => :string, :default => ENV["INGRESS_API"] || "http://localhost:9292"
     opt :config, "Configuration file name", :type => :string, :default => ENV["CONFIG"] || "default"
     opt :amounts, "Amount of generated items", :type => :string, :default => ENV["AMOUNTS"] || "default"
+    opt :configmap, "Config in YAML format", :type => :string, :default => ENV["ENDPOINT_CONFIG"]
```
Comment (carbonin) [inline bin/mock-collector]: I don't think this line makes sense.

A configmap is usually provided as a file mounted into the pod. If you want a string, don't call it a configmap.

On the other hand, how big is this going to be? I would rather provide a configmap for a larger payload then make this parameter a file name. I thought this was the original intent of the `config` parameter.

**Review by carbonin — CHANGES_REQUESTED** [review]
Can you explain why we need a new parameter for this? Why wouldn't we mount a configmap up in one or both of the existing locations?

File: bin/mock-collector (hunk 1 above)
Comment (slemrmartin) [reply bin/mock-collector]: @carbonin agree. I don't know if the ENV size limit is 32 or 128kB, but it can be more. Closing this PR

**Pull request closed without merging**